docs/catalog.md

```diff
@@ -0,0 +1,76 @@
+`catalog.yaml`
+
+# Abstract
+
+Describes what the file `catalog.yaml` contains and the use-cases indented for this **contract**. The file will be placed on the root of a CVS repository containing Tekton Pipeline resources (Tasks and Pipelines).
+
+The `catalog.yaml` goal is to serve as a blueprint to find the resources managed on the respective repository, as well to provide information for software supply chain attestation, and describe continuous integration test-cases.
+
+During the release of these repositories the `catalog.yaml` is added to the payload in order to describe the Tekton resource artifacts.
+
+# Use-Cases
+
+The file described on this document is meant to make possible the use-cases described below.
+
+## Repository Root
+
+The primary location for the `catalog.yaml` file is on the root of the (Git?) repository, describing all the elements, providing software supply chain attestation data and as well descring test cases.
+
+For repositories containing the direct YAML payload of Tekton resource files stored, the file will also contain `.catalog.resources` entries, reflecting the location of the data.
+
+### Release Artifacts
+
+The `catalog.yaml` should be present on the repositories release payload, therefore when the maintainers decide to release a new version, the `catalog.yaml` is able to overwrite the entries on `.catalog.resources`.
+
+This ability makes possible to template Tekton resources instead of the plain YAML files, and during the regular releases the resource are assembled.
+
+## Continuous Integration
+
+# `catalog.{yml,yaml}`
+
+The file looks like the example below.
+
+```yml
+---
+version: v1
+
+catalog:
+  repository:
+    description: Tekton Task to interact with Git repositories
+  attestation:
+    publicKey: path/to/public.key
+    annotations:
+      team: tekton-ecosystem
+  resources:
+    tasks:
+      - name: task-git
+        version: "0.0.1"
+        filename: path/to/resource.yaml
+        sha256sum: resource-sha256-checksum
+        tri: git://openshift-pipelines/task-git@0.0.1
+        bundle: ghcr.io/openshift-pipelines/task-git:0.0.1-bundle
+        signature: path/to/signature.sig
```

Comment:
not sure these entry are latest or not

because when i download catalog.yaml from task-containers repo i see 

```
      - name: buildah
        version: 0.2.0
        filename: tasks/task-buildah/task-buildah.yaml
        checksum: 012e1cfb4c8e857d6da2b8d382e30fe1f76b00662e505ca0ec93853ec0c3aa5d
        signature: ""
```

So just cross checking

Reply:
Same as above, I need to refine that a bit, it's a slightly out of sync